test/powershell/Host/ConsoleHost.Tests.ps1

```diff
@@ -226,6 +226,23 @@ Describe "ConsoleHost unit tests" -tags "Feature" {
             & $powershell $testdrive/$Filename
             $LASTEXITCODE | Should -Be 123
         }
+
+        It "A single dash should be passed as an arg" {
+            $testScript = @'
+    [CmdletBinding()]param(
+        [string]$p1,
+        [string]$p2,
+        [Parameter(ValueFromPipeline)][string]$InputObject
+    )
+    process{
+        $input.replace($p1, $p2)
+    }
+'@
+            $testFilePath = Join-Path $TestDrive "test.ps1"
+            Set-Content -Path $testFilePath -Value $testScript
+            $observed = echo hello | pwsh $testFilePath e -
```

Comment:
I'd add the explicit check:
```suggestion
            $observed = echo hello | pwsh -File $testFilePath e -
```

It seems we haven't test for `pwsh $testFilePath -`. Comment in code says that it should redirect input.

Reply:
We are testing the case that broke here, which was a non-redirection case.  

The redirection cases are here:  https://github.com/PowerShell/PowerShell/blob/43e4e31044b812f69c981b417038df208fd0cbff/test/powershell/Host/ConsoleHost.Tests.ps1#L441-L453

Reply:
The test doesn't seems work because follow fail:
```
 pwsh C:\Users\1\Documents\UsingSort.ps1 -
C:\Users\1\Documents\UsingSort.ps1 : Cannot process argument because the value of argument "name" is not valid. Change the value of the "name" argument and run the operation again.
+ CategoryInfo          : InvalidArgument: (:) [UsingSort.ps1], PSArgumentException
+ FullyQualifiedErrorId : Argument,UsingSort.ps1
`

Reply:
No, this is not how this works.  If you specify a filename, the `-` is a parameter to your script and this is expected.  Re-closing the issue.